backend_plugin/src/evaluators/Evaluator.java

```diff
@@ -51,10 +53,12 @@ private void initializeListeners(IEditorPart editorWindow) {
 		// Get the document stored inside the editor window
 		ITextEditor editor = (ITextEditor) editorWindow;
```

Comment:
I'm a little nervous of this cast without checking if `editor isinstanceof ITextEditor`. Also, since this method seems to work exclusively on `ITextEditors`, it's probably worth it to do this cast in the constructor an then make `initializeListeners` take an `ITextEditor` unless you anticipate this method being called from inside some native eclipse handler or somewhere else that can only pass an IEditorPart.